src/libraries/Common/src/System/TimeProvider.cs

```diff
@@ -53,9 +61,9 @@ public DateTimeOffset LocalNow
                 TimeSpan offset = LocalTimeZone.GetUtcOffset(utcDateTime);
 
                 long localTicks = utcDateTime.Ticks + offset.Ticks;
-                if ((ulong)localTicks > DateTime.MaxTicks)
+                if ((ulong)localTicks > (ulong)DateTime.MaxValue.Ticks)
                 {
-                    localTicks = localTicks < DateTime.MinTicks ? DateTime.MinTicks : DateTime.MaxTicks;
+                    localTicks = localTicks < DateTime.MinValue.Ticks ? DateTime.MinValue.Ticks : DateTime.MaxValue.Ticks;
                 }
```

Comment:
Is there any performance concern? It seems that `DateTime.MaxValue.Ticks` isn't optimized to constant.

Reply:
fixed it. Thanks!